### ISSUE TITLE
NanoAODv6 randomized screening production issue

### ISSUE THREAD

**Issue by cedricpri** [cms-nanoAOD/cmssw#503]
Our signal samples for our analysis have been produced up to the nanoAODv5, but for some reason it appears that the central production of nanoAODv6 crashed, as far as we can see in McM [1].

In order to understand this issue, we tried to process the nanoAOD ourselves using CMSSW_10_2_18 and we ran into the same issue, when using the configuration found in McM with the option --eventcontent NANOAODSIM in order to produce a flat tree. The error we got in this case is [2].

Clearly, it seems that the nanoAOD code attempts to create a branch called "GenModel_"+<mass_point>, but in our particular case the <mass_point> seems to contain forbidden characters, such as "/" and "-" (which gives us issues when trying to plot these variables later on).

We manage to fix this issue but removing such characters from the variables defined in [3, 4], but a central fix might be useful in case other groups run into the same issue.

Thanks!

[1] https://cms-pdmv.cern.ch/mcm/requests?page=0&dataset_name=*TbarDMJets_Dilepton_pseudoscalar_LO_TuneCP5_13TeV-madgraph-mcat*&member_of_campaign=*17*&shown=127
[2] Extra characters after type tag 'TTbarDMJets_Dilepton_pseudoscalar_LO_Mchi-20_Mphi-100_TuneCP5_13TeV-madgraph-mcatnlo-pythia8/O' for branch 'GenModel_/TTbarDMJets_Dilepton_pseudoscalar_LO_Mchi-20_Mphi-100_TuneCP5_13TeV-madgraph-mcatnlo-pythia8'; must be one character.
 [3] https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/plugins/GenWeightsTableProducer.cc#L272
 [4] https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/plugins/GenWeightsTableProducer.cc#L801

**Comment by peruzzim** [comment]
Thank you for reporting, we definitely have to fix this.

**Comment by peruzzim** [comment]
fixed in #505

e.g.
```
*Br 1449 :GenModel__TTbarDMJets_Dilepton_pseudoscalar_LO_Mchi_1_Mphi_250_TuneCP5_13TeV_madgraph_mcatnlo_pythia8 : *
*         | Bool_t EventString bit                                           *
*Entries :      300 : Total  Size=       1245 bytes  File Size  =        251 *
*Baskets :        1 : Basket Size=      32000 bytes  Compression=   1.87     *
*............................................................................*
*Br 1450 :GenModel__TTbarDMJets_Dilepton_pseudoscalar_LO_Mchi_1_Mphi_450_TuneCP5_13TeV_madgraph_mcatnlo_pythia8 : *
*         | Bool_t EventString bit                                           *
*Entries :      300 : Total  Size=       1245 bytes  File Size  =        251 *
*Baskets :        1 : Basket Size=      32000 bytes  Compression=   1.87     *
```
